controller/volume_controller.go

```diff
@@ -821,10 +825,13 @@ func (vc *VolumeController) ReconcileVolumeState(v *longhorn.Volume, es map[stri
 		return err
 	}
 
-	if err := vc.checkForAutoAttachment(v, e, rs, allScheduled); err != nil {
+	hasEvictionRequestedReplicas := vc.hasReplicaEvictionRequested(rs)
+
+	if err := vc.checkForAutoAttachment(v, e, rs, allScheduled, hasEvictionRequestedReplicas); err != nil {
```

Comment:
Both auto-attach and detach function can have replicas as a parameter, so we don't need to check the condition outside then pass it in.

Reply:
then the new function will be called twice. Each once in auto-attach and auto-detach, is that ok?

Reply:
That's fine since it's just a status check, doesn't have side effects.

Reply:
Ok, will fix it.

Reply:
Sorry my bad. auto-detach doesn't have replicas as parameter passed in. Should I add the rs new parameter or keep this way? Thanks.

Reply:
We can just add `rs` parameter. It's still a small change.